api/turing/api/router_versions_api.go

```diff
@@ -33,6 +34,47 @@ func (c RouterVersionsController) ListRouterVersions(
 	return Ok(routerVersions)
 }
 
+// CreateRouterVersion creates a router version from the provided configuration. If no router exists
+// within the provided project with the provided id, this method will throw an error.
+// If the update is valid, a new RouterVersion will be created but NOT deployed.
+func (c RouterVersionsController) CreateRouterVersion(r *http.Request, vars RequestVars, body interface{}) *Response {
+	// Parse request vars
+	var errResp *Response
+	var router *models.Router
+	if _, errResp = c.getProjectFromRequestVars(vars); errResp != nil {
+		return errResp
+	}
+	if router, errResp = c.getRouterFromRequestVars(vars); errResp != nil {
+		return errResp
+	}
+
+	requestConfig := body.(*request.RouterConfig)
+
+	// This is used temporarily in order to reuse the BuildRouterVersion method
+	dummyRouterRequest := request.CreateOrUpdateRouterRequest{
+		Environment: "",
```

Comment:
Even though we're using `RouterConfig` without `EnvironmentName` and (Router) `Name` as the request body, there's a need to construct this dummy `request.CreateOrUpdateRouterRequest` object because the existing `BuildRouterVersion` struct method is defined only for the said struct.

In the future when the refactoring of all the controller/service methods are done, this dummy object can be removed.